### DISCUSSION TITLE
Cloning PICGenios to do some modifications

### DISCUSSION THREAD

**Discussion by eagl1** [lcgamboa/picsimlab#62] [Ideas]
Hi,

Is there a possibility to clone PICGenios ?

I'm an electronics diploma trainer in a technical college, Saudi Arabia. I started working with picsimlab last semester and I liked it, and wish to do some modifications to fit my applications I want to work on.

I'm thinking of the following:
1. Do some modifications on PICGenios.
2. Do similar applications to the Arduino one, which is to do application modules to interface with the PIC development board; like, water level control system, 4-way traffic system, temperature control system, motor control system, servo, stepper and DC motors experiments, IR modules, Bluetooth, etc.

I know it would be a lot of work, and I'm willing to work and contribute to this wonderful project.

**Comment by lcgamboa** [comment]
Hi,
I don't know exactly what you mean by "clone", but as it is an open source project I believe it is possible to make the changes you intend.

The PICGenios board is one of the most complex, using the breadboard board with the same configuration as the PICGenios allows making modifications without having to change the code. Take a look at this [example](https://lcgamboa.github.io/picsimlab_examples/board_Breadboard.html#board_Breadboard_PIC18F4550_test_b0).

There is already a simulation of a liquid tank, as described in the [documentation](https://lcgamboa.github.io/picsimlab_docs/stable/SerialRemoteTank.html#x47-460008.2). 
It is also possible to simulate [temperature control system](https://lcgamboa.github.io/picsimlab_docs/stable/TemperatureSystem.html#x105-11200010.4.12), [servo](https://lcgamboa.github.io/picsimlab_docs/stable/ServoMotor.html#x91-9800010.3.14), [stepper](https://lcgamboa.github.io/picsimlab_docs/stable/StepMotor.html#x92-9900010.3.15) and [DC motors](https://lcgamboa.github.io/picsimlab_docs/stable/DCMotor.html#x81-8800010.3.4).

IR modules and Bluetooth are not supported yet, but there are the the ethernet shield w5500 part and the experimental version of ESP32 has WIFI simulation.

PICSimLab has an interface for [remote control](https://lcgamboa.github.io/picsimlab_docs/stable/RemoteControlInterface.html#x15-140003.4) through a TCP/IP socket. It's the simplest way to interface with other simulators. There are some (poorly documented) examples of [PICSimLab integration with Webots](https://github.com/lcgamboa/picsimlab_rcontrol). 
Maybe it's a simpler way to interact with a 4-way traffic system simulator.

Happy hacking

**Reply by eagl1** [reply]
Hi,

Thank you so much for your reply and the wonderful answer.

I'm really impressed with all the features you mentioned, how it's an open source and also how it's possible to do modifications.

Up to now I think there are a lot for me to learn about the board, and then the add on applications you mentioned.

Actually the most important modification I want to do is to separate the 8 LEDs connected to port D from sharing the signals with the 7-segments and the LCDs lines. Because I want to use them to do control signals beside controlling the 7-segment and the LCDs without affecting the state of the LEDs
